dune/iga/geometrykernel/nurbspatchgeometry.hh

```diff
@@ -317,6 +317,102 @@ namespace Dune::IGANEW::GeometryKernel {
+     *
+     * @tparam boundary The boundary index to check.
+     * @tparam tol Absolute tolerance for comparing control points. Default is 1e-8.
+     *
+     * @throws Dune::NotImplemented If the dimension is greater than 3 (higher dimensions not implemented yet).
+     */
+    ConnectionResult isConnectedAtBoundary(int boundary, double tol = 1e-8) const {
+      const auto& cps = patchData_.controlPoints;
+      if constexpr (mydimension == 1) {
+        if (auto cp = cps.directGetAll();
+            FloatCmp::eq<FieldVector<ctype, coorddimension>, FloatCmp::CmpStyle::absolute>(cp.front().p, cp.back().p,
+                                                                                           tol))
+          return ConnectionResult(true, !boundary);
+        else
+          return ConnectionResult(false, -1);
+      } else if constexpr (mydimension == 2) {
+        auto edgeRange = [&cps](int b) {
+          switch (b) {
+            case 0:
+              return cps.hyperSurfFront({1});
+            case 1:
+              return cps.hyperSurfBack({1});
+            case 2:
+              return cps.hyperSurfFront({0});
+            case 3:
+              return cps.hyperSurfBack({0});
+            default:
+              assert("No valid boundary index");
+          }
+          __builtin_unreachable();
+        };
+
+        auto theEdgeToCheck = edgeRange(boundary);
+        for (auto i : Dune::range(4)) {
+          if (boundary == i) continue;
+          if (std::ranges::equal(theEdgeToCheck, edgeRange(i), [tol](auto f, auto s) {
+                return FloatCmp::eq<FieldVector<ctype, coorddimension>, FloatCmp::CmpStyle::absolute>(f.p, s.p, tol);
+              }))
+            return ConnectionResult(true, i);
+        }
+        return ConnectionResult(false, -1);
+      } else if constexpr (mydimension == 3) {
+        auto edgeRange = [&cps](int b) {
+          switch (b) {
+            case 0:
+              return cps.hyperSurfFront({1, 2});
+            case 1:
+              return cps.hyperSurfBack({1, 2});
+            case 2:
+              return cps.hyperSurfFront({0, 1});
+            case 3:
+              return cps.hyperSurfBack({0, 1});
+            case 4:
+              return cps.hyperSurfFront({0, 2});
+            case 5:
+              return cps.hyperSurfBack({0, 2});
+            default:
+              assert("No valid boundary index");
+          }
+          __builtin_unreachable();
+        };
+
+        auto theEdgeToCheck = edgeRange(boundary);
+        for (auto i : Dune::range(6)) {
+          if (boundary == i) continue;
+          if (std::ranges::equal(theEdgeToCheck, edgeRange(i), [tol](auto f, auto s) {
+                return FloatCmp::eq<FieldVector<ctype, coorddimension>, FloatCmp::CmpStyle::absolute>(f.p, s.p, tol);
+              }))
+            return ConnectionResult(true, i);
+        }
+        return ConnectionResult(false, -1);
+      } else
+        DUNE_THROW(Dune::NotImplemented,
```

Comment:
Nice addition 👍 
Regarding the error handling: 
Couldn't we just do 
```cpp
 ConnectionResult isConnectedAtBoundary(int boundary, double tol = 1e-8) const  requires (dim <= 3) {
 }
 ```
 and then 
 ```cpp
 ConnectionResult isConnectedAtBoundary(int boundary, double tol = 1e-8) const  {
  DUNE_THROW(Dune::NotImplemented)
 }
 ```

Reply:
But maybe its a bit to verbose to have 2 function definitions

Reply:
The result would be the same, though. I think the current version is less verbose. But we can change the runtime error to a compile time error by just add `requires (dim <= 3)`, since the dimension is known most of the time anyway.